rs/orchestrator/src/catch_up_package_provider.rs

```diff
@@ -173,14 +179,16 @@ impl CatchUpPackageProvider {
             );
             None
         })?;
-        let url = http_endpoint_to_url(&http, &self.logger)?;
+        let uri = http_endpoint_to_url(&http, &self.logger)?.to_string() + "_/catch_up_package";
```

Comment:
Can you use `path_segments_mut` here such that we avoid accidentally missing `/` or having `//`